tests/graph/benches/graph/scenario/stages/entity_queries.rs

```diff
@@ -0,0 +1,137 @@
+use core::error::Error;
+
+use error_stack::{Report, ResultExt as _};
+use hash_graph_store::{
+    entity::{EntityQuerySorting, EntityStore as _, GetEntitiesParams},
+    filter::Filter,
+    pool::StorePool as _,
+    subgraph::temporal_axes::QueryTemporalAxesUnresolved,
+};
+use hash_graph_test_data::seeding::producer::ontology::WebCatalog as _;
+use type_system::principal::actor::{ActorEntityUuid, ActorId, UserId};
+
+use super::Runner;
+
+#[derive(Debug, derive_more::Display)]
+pub enum QueryEntitiesError {
+    #[display("Unknown user catalog: {name}")]
+    UnknownUserCatalog { name: String },
+    #[display("Catalog should not be empty")]
+    EmptyCatalog,
+    #[display("Failed to acquire database connection")]
+    Acquire,
+    #[display("Failed to query entities")]
+    Query,
+}
+
+impl Error for QueryEntitiesError {}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct QueryEntitiesInputs {
+    #[serde(default)]
+    pub user_catalog: Option<String>,
+    pub limit: Option<usize>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct QueryEntitiesByUserStage {
+    pub id: String,
+    pub inputs: QueryEntitiesInputs,
+    #[serde(default)]
+    pub stage_id: Option<u16>,
+}
+
+#[derive(Debug, Default, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct QueryEntitiesByUserResult {
+    pub num_entities: usize,
+    pub actor: Option<ActorId>,
+}
+
+impl QueryEntitiesByUserStage {
+    pub async fn execute(
+        &self,
+        runner: &mut Runner,
+    ) -> Result<QueryEntitiesByUserResult, Report<QueryEntitiesError>> {
+        let user_catalog = self
+            .inputs
+            .user_catalog
+            .as_ref()
+            .map(|key| {
+                runner.resources.user_catalogs.get(key).ok_or_else(|| {
+                    Report::new(QueryEntitiesError::UnknownUserCatalog { name: key.clone() })
+                })
+            })
+            .transpose()?;
+
+        let actor = user_catalog
+            .map(|catalog| {
+                let num_users = catalog.len();
+                if num_users == 0 {
+                    return Err(Report::new(QueryEntitiesError::EmptyCatalog));
+                }
+                // TODO: Add diverse user selection strategies (heavy/light users, random,
+                //       permission-based)
+                #[expect(
+                    clippy::float_arithmetic,
+                    clippy::cast_possible_truncation,
+                    clippy::cast_sign_loss,
+                    clippy::cast_precision_loss,
+                    reason = "https://linear.app/hashintel/issue/BE-31"
+                )]
+                let user_idx = ((num_users as f64 * 0.5) as usize).clamp(0, num_users - 1);
```

Comment:
I am guessing this is `* 0.5` to prepare for other strategies? Otherwise `/ 2` would be equivalent no?